requests/src/requests/Requester.scala

```diff
@@ -230,7 +230,17 @@ case class Requester(verb: String,
         for((k, v) <- compress.headers) connection.setRequestProperty(k, v)
 
         connection.setReadTimeout(readTimeout)
-        auth.header.foreach(connection.setRequestProperty("Authorization", _))
+
+        auth match {
+          case basic: RequestAuth.Basic =>
+            connection.setRequestProperty("Authorization", basic.credentials.get)
+          case bearer: RequestAuth.Bearer =>
+            connection.setRequestProperty("Authorization", bearer.credentials.get)
```

Comment:
Line 236 and 238 are doing the same thing, maybe we can use `case Foo | Bar` to handle them both together in the same `case`.

Reply:
I've decided to tweak my implementation. Would appreciate any feedback you have on the latest round of changes :)